ros/processor/insights_engine_result_consumer.py

```diff
@@ -66,8 +66,13 @@ def handle_msg(self, msg):
                 for report in reports:
                     if 'cloud_instance_ros_evaluation' in report["rule_id"]:
                         ros_reports.append(report)
-
-                self.process_report(host, ros_reports)
+                if len(ros_reports) == 0:
+                    with app.app_context():
+                        deleted_id = delete_record(db.session, System, 'inventory_id',
+                                                   inventory_id=host['id'])
+                        LOG.info("Deleted system with inventory_id: %s via engine-result", deleted_id)
```

Comment:
```suggestion
                        inventory_id = delete_record(db.session, System, 'inventory_id',
                                                   inventory_id=host['id'])
                        LOG.info("Deleted system with inventory_id: %s via engine-result", inventory_id)
```


Any impact after removing a whole record from System table? I guess it will recreate System record next time.

Reply:
Yeah, no such impact. It would create a new record next time.

Reply:
Marked this comment as `Unresolved` because having concern around system record deletion change instead of clearing fields.  Currently ROS DB is the only place to get info about for how many systems ROS is enabled.